examples/z_non_blocking_get.c

```diff
@@ -58,9 +58,9 @@ int main(int argc, char **argv) {
         }
         if (z_reply_is_ok(&reply)) {
             z_sample_t sample = z_reply_ok(&reply);
-            char *keystr = z_keyexpr_to_string(sample.keyexpr);
-            printf(">> Received ('%s': '%.*s')\n", keystr, (int)sample.payload.len, sample.payload.start);
-            free(keystr);
+            z_owned_str_t keystr = z_keyexpr_to_string(sample.keyexpr);
+            printf(">> Received ('%s': '%.*s')\n", z_loan(keystr), (int)sample.payload.len, sample.payload.start);
+            z_drop(&keystr);
```

Comment:
In Zenoh-Pico, we have it implemented as `z_drop(z_move(keystr))`.
I know it might seem redundant but by doing so we dont break the rule of being explicit when moving things to the Zenoh API.